src/OpenTelemetry.Exporter.Geneva/Metrics/GenevaMetricExporter.cs

```diff
@@ -22,31 +23,46 @@ public class GenevaMetricExporter : BaseExporter<Metric>
 
     internal const string DimensionKeyForCustomMetricsNamespace = "_microsoft_metrics_namespace";
 
-    private readonly TlvMetricExporter exporter;
+    private readonly IDisposable exporter;
 
-    private delegate ExportResult ExportMetricsFunc(in Batch<Metric> batch);
+    private delegate ExportResult ExportMetricsFunc(in Batch<Metric> batch, Resource resource);
 
     private readonly ExportMetricsFunc exportMetrics;
 
     private bool isDisposed;
 
+    private Resource resource;
+
+    internal Resource Resource => this.resource ??= this.ParentProvider.GetResource();
```

Comment:
Make this an `internal readonly` field and initialize it in the ctor?

Reply:
Since this is only required by the new exporter, could you just pass the resource to its ctor instead of updating the Export method signature for TlvMetricExporter as well.

Reply:
resource is not available before the .Build(). it gets set only after the exporter is initialized.

Reply:
> Since this is only required by the new exporter, could you just pass the resource to its ctor instead of updating the Export method signature for TlvMetricExporter as well.

Resource is only needed by `OtlpProtobufMetricExporter` and need not be shared with `TlvMetricExporter`.

Reply:
`ParentProvider` is also set after the processor is created so this won't work.

Reply:
There are a few ways to address this issue. One would be to let the `OtlpProtobufExporter` have a field that pointes to `GenevaMetricExporter` or you could provide `OtlpProtobufExporter` a `Func` that would return the Resource. How about this?

```csharp
public class GenevaMetricExporter : BaseExporter<Metric>
{
    public GenevaMetricExporter(GenevaMetricExporterOptions options)
    {
        ...

        var otlpProtobufExporter = new OtlpProtobufMetricExporter(this.GetResource);

        ...
    }

    private Resource GetResource(
    {
        return this.ParentProvider.GetResource();
    }
}




internal sealed class OtlpProtobufMetricExporter : IDisposable
{
    private readonly Func<Resource> getResource;
    private Resource resource;
    private Resource Resource => this.resource ??= this.getResource();

    public OtlpProtobufMetricExporter(Func<Resource> getResource)
    {
        ...

        this.getResource = getResource;

        ...
    }

    public ExportResult Export(in Batch<Metric> batch)
    {
        ...

        try
        {
            this.otlpProtobufSerializer.SerializeAndSendMetrics(this.buffer, ref currentPosition, this.Resource, batch);
        }
        catch (Exception ex)
        {
            ...
        }

        return result;
    }
}
```